eos/effects/adaptivearmorhardener.py

```diff
@@ -2,17 +2,207 @@
-            damagePatternModifier = getattr(damagePattern,attrDamagePattern)/float(sum((damagePattern.emAmount,damagePattern.thermalAmount,damagePattern.kineticAmount,damagePattern.explosiveAmount)))
-            modifiedResistModifier = (1-(((1-module.getModifiedItemAttr(attr))*4)*(damagePatternModifier)))
-            module.forceItemAttr(attr, modifiedResistModifier)
-
-        fit.ship.multiplyItemAttr(attr, module.getModifiedItemAttr(attr), stackingPenalties=True, penaltyGroup="preMul")
+    damagePattern = fit.damagePattern
+
+    # Skip if there is no damage pattern. Example: projected ships or fleet boosters
+    if damagePattern:
+
+        # Populate a tuple with the damage profile modified by current armor resists.
+        # Build this up front as there are a number of values we only need to touch once
+        damagePattern_tuple = []
+
+        '''
+        # Bit of documentation for how the tuple is layed out
+        damagePattern_tuple.append(['RESIST TYPE',
+                SHIP ARMOR RESIST AMOUNT,
+                DAMAGE PATTERN AMOUNT,
+                MODIFIED DAMAGE PATTERN AMOUNT (SHIP ARMOR RESIST AMOUNT * DAMAGE PATTERN AMOUND),
+                ADAPTIVE RESIST AMOUNT,
+                ADAPTIVE RESIST AMOUNT (ORIGINAL))
+        '''
+
+        damagePattern_tuple.append(['Em',
+                fit.ship.getModifiedItemAttr('armorEmDamageResonance'),
+                damagePattern.emAmount,
+                damagePattern.emAmount*fit.ship.getModifiedItemAttr('armorEmDamageResonance'),
+                module.getModifiedItemAttr('armorEmDamageResonance'),
+                module.getModifiedItemAttr('armorEmDamageResonance')])
+
+        damagePattern_tuple.append(['Thermal',
+                fit.ship.getModifiedItemAttr('armorThermalDamageResonance'),
+                damagePattern.thermalAmount,
+                damagePattern.thermalAmount * fit.ship.getModifiedItemAttr('armorThermalDamageResonance'),
+                module.getModifiedItemAttr('armorThermalDamageResonance'),
+                module.getModifiedItemAttr('armorThermalDamageResonance')])
+
+        damagePattern_tuple.append(['Kinetic',
+                fit.ship.getModifiedItemAttr('armorKineticDamageResonance'),
+                damagePattern.kineticAmount,
+                damagePattern.kineticAmount*fit.ship.getModifiedItemAttr('armorKineticDamageResonance'),
+                module.getModifiedItemAttr('armorKineticDamageResonance'),
+                module.getModifiedItemAttr('armorKineticDamageResonance')])
+
+        damagePattern_tuple.append(['Explosive',
+                fit.ship.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                damagePattern.explosiveAmount,
+                damagePattern.explosiveAmount*fit.ship.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                module.getModifiedItemAttr('armorExplosiveDamageResonance'),
+                module.getModifiedItemAttr('armorExplosiveDamageResonance')])
```

Comment:
It's not a big deal, but initially sorting the damage types Em, Thermal, Kinetic, Explosive causes incorrect sorting of equal values later. The game seems to sort Em, Explosive, Kinetic, Thermal.